symengine/polynomial.cpp

```diff
@@ -73,57 +56,49 @@ int UnivariatePolynomial::compare(const Basic &o) const
     return map_uint_mpz_compare(dict_, s.dict_);
 }
 
-RCP<const Basic> UnivariatePolynomial::from_dict(const RCP<const Symbol> &var, map_uint_mpz &&d)
-{
+RCP<const Basic> UnivariateIntPolynomial::from_dict(const RCP<const Symbol> &var, map_uint_mpz &&d) {
     if (d.size() == 1) {
         if (d.begin()->first == 0)
             return integer(d.begin()->second);
-        else if (d.begin()->first == 1) {
+        else if (d.begin()->first == 1) 
             if (d.begin()->second == 0)
                 return zero;
             else if (d.begin()->second == 1)
                 return var;
             else
                 return Mul::from_dict(integer(d.begin()->second), {{var, one}});
-        } else {
-            if (d.begin()->second == 0)
-                return zero;
-            else if (d.begin()->second == 1)
-                return pow(var, integer(d.begin()->first));
-            else
-                return Mul::from_dict(integer(d.begin()->second),
-                    {{var, integer(d.begin()->first)}});
-        }
-    } else {
-        return make_rcp<const UnivariatePolynomial>(var, (--(d.end()))->first, std::move(d));
-    }
+        else if (d.begin()->second == 0)
+            return zero;
+        else if (d.begin()->second == 1)
+            return pow(var, integer(d.begin()->first));
+        else
+            return Mul::from_dict(integer(d.begin()->second), {{var, integer(d.begin()->first)}});
+    } else 
+        return make_rcp<const UnivariateIntPolynomial>(var, (--(d.end()))->first, std::move(d));
```

Comment:
If I only have that one line in from_dict I get a segmentation violation
[errorpoly.txt](https://github.com/symengine/symengine/files/159684/errorpoly.txt)